qiskit/transpiler/passes/optimization/commutative_cancellation.py

```diff
@@ -120,7 +150,10 @@ def run(self, dag):
 
                 # Replace the data of the first node in the run
                 if cancel_set_key[0] == 'z_rotation':
-                    new_op = U1Gate(total_angle)
+                    if var_z_gate:
+                        new_op = var_z_gate(total_angle)
```

Comment:
Do you know if this pass currently preserves global phase? If so, is there any global phase correction needed for choosing say Rz vs p?

Reply:
It does not currently preserve global phase.

Reply:
I'll correct for phase in this pr.